File: API/Shipments/Models/ShippedCollection.cs
namespace Shipments.Models;

public class ShippedCollection
{
    public int CollectionId { get; set; }
    public int CollectionCenterCorporateId { get; set; }
    public int MerchantCorporateId { get; set; }
    public int TransporterCorporateId { get; set; }
    public int FarmerId { get; set; }
    public string? CropName { get; set; }
    public string? VehicleNumber { get; set; }
    public string? Grade { get; set; } 
    public string? ContainerType { get; set; }
    public int Quantity { get; set; }
    public double TotalWeight { get; set; }
    public double NetWeight { get; set; }
    public DateTime CollectionDate { get; set; }
    public DateTime ShipmentDate { get; set; }

}

File: UI/EAgroUIApp/src/app/collectioncenter/shipped-collection.ts
export class ShippedCollection {
    constructor(
        public collectionId: number,
        public merchantCorporateId: number,
        public merchantName: string,
        public collectionCenterCorporateId: number,
        public collectionCenterName: string,
        public transporterCorporateId: number,
        public transporteName: string,
        public farmerId: number,
        public farmerName: string,
        public cropName: string,
        public vehicleNumber: string,
        public grade: string,
        public containerType: string,
        public quantity: number,
        public totalWeight: number,
        public netWeight: number,
        public collectionDate: string,
        public shipmentDate: string
    ) { }
}
